src/components/SwipeableView/index.native.js

```diff
@@ -1,39 +1,109 @@
-import React, {PureComponent} from 'react';
-import {PanResponder, View} from 'react-native';
+import React, {Component} from 'react';
+import {
+    PanResponder, View, Dimensions, Animated,
+} from 'react-native';
 import PropTypes from 'prop-types';
+
+import styles from '../../styles/styles';
 import CONST from '../../CONST';
 
 const propTypes = {
     children: PropTypes.element.isRequired,
 
     /** Callback to fire when the user swipes down on the child content */
-    onSwipeDown: PropTypes.func.isRequired,
+    onSwipeDown: PropTypes.func,
+
+    /** Callback to fire when swiping left or right */
+    onSwipeHorizontal: PropTypes.func,
+
+    /** These help to prevent a swipe animation when at either end */
+    canSwipeLeft: PropTypes.bool,
+    canSwipeRight: PropTypes.bool,
+
+    /** Callback to facility an press event */
+    onPress: PropTypes.func,
+
```

Comment:
Similarly here too!

Reply:
I changed this comment to explain that I need it here to handle the onPress event because panResponder takes in all gestures when it's set